### PR TITLE
feat(desktop): implement S01 desktop-controlled symphony runtime lifecycle

### PR DESCRIPTION
**Pull request by gannonh**
## Summary
- deliver S01 runtime lifecycle plumbing across Desktop main/preload/renderer for Symphony start/stop/restart management
- add deterministic runtime config resolution and launch descriptor handling (URL/workflow path/binary discovery)
- add supervisor + IPC + renderer runtime controls and status surfaces with lifecycle/error visibility
- add e2e runtime lifecycle coverage and smoke checklist documentation
- harden branch/coverage with additional symphony config and supervisor test cases

## Validation
- `cd apps/desktop && npx vitest run src/main/__tests__/symphony-config.test.ts src/main/__tests__/symphony-supervisor.test.ts src/main/__tests__/symphony-ipc.test.ts`
- `cd apps/desktop && bun run typecheck`
- `cd apps/desktop && npx playwright test e2e/tests/symphony-runtime.e2e.ts`
- `cd apps/desktop && bun run test`

## Summary by CodeRabbit

* **New Features**
  * Symphony runtime status indicator added to app header.
  * New "Symphony" settings panel to view runtime details and Start/Stop/Restart the runtime.
  * Real-time status display showing phase, PID, URL, diagnostics and actionable error guidance for configuration issues.

* **Tests**
  * Added end-to-end and unit tests covering lifecycle, config error, readiness failure, and control behaviors.

<h3>Greptile Summary</h3>

This PR delivers the S01 Symphony runtime lifecycle plumbing for Kata Desktop, wiring together a `SymphonySupervisor` (child-process management with SIGTERM/SIGKILL stop, readiness polling, and diagnostic capture), deterministic config resolution (`symphony-config.ts`), IPC handlers for start/stop/restart/status, a preload bridge, Jotai state atoms, and a `SymphonyRuntimePanel` UI with lifecycle controls and error surfaces. It also adds a full E2E test suite with a `symphonyMockMode` fixture and extensive unit tests for config and supervisor logic.

**Key changes:**
- `SymphonySupervisor` manages the Symphony child process end-to-end, including readiness polling against `/api/v1/state`, graceful stop with SIGTERM/SIGKILL fallback, and a `mockMode` path for E2E testing.
- `symphony-config.ts` resolves binary path (env override → bundled → PATH), URL (preferences → `KATA_SYMPHONY_URL` → `SYMPHONY_URL`), and workflow path (preferences → default `WORKFLOW.md`) with deterministic error codes per failure mode.
- IPC handlers in `ipc.ts` gate all symphony commands through `runSymphonyCommand` and push status events to the renderer via `webContents.send`.
- `before-quit` in `index.ts` properly stops the supervisor before bridge shutdown.
- Two issues found: env-var constant names in `symphony-config.ts` are inverted (minor naming confusion), and in `stopInternal` the SIGKILL outcome is never checked — the supervisor unconditionally returns `success: false` even when SIGKILL successfully terminates the process, leaving the UI in a `Failed` state after a clean forceful kill.

<h3>Confidence Score: 3/5</h3>

Safe to merge with the SIGKILL outcome bug acknowledged — it produces a misleading failure state but does not corrupt data or leave zombie processes.

The PR is well-structured with comprehensive test coverage (unit + E2E + mock modes). The SIGKILL outcome not being checked is a real logic gap: after a graceful-stop timeout, SIGKILL typically succeeds, but the supervisor always transitions to phase 'failed' regardless, giving the user a false error state. The env-var naming inversion is a minor maintenance hazard. No data-loss, security, or correctness-critical issues found elsewhere.

apps/desktop/src/main/symphony-supervisor.ts — stopInternal SIGKILL path; apps/desktop/src/main/symphony-config.ts — constant naming

<details><summary><h3>Important Files Changed</h3></summary>

| Filename | Overview |
|----------|----------|
| apps/desktop/src/main/symphony-supervisor.ts | New supervisor class managing Symphony child-process lifecycle; SIGKILL outcome is not checked in stopInternal, causing a spurious 'failed' state after successful forceful termination. |
| apps/desktop/src/main/symphony-config.ts | Config resolution for Symphony launch; inverted constant names (FALLBACK/DEFAULT) are misleading about priority ordering, but logic is functionally correct. |
| apps/desktop/src/main/ipc.ts | Symphony IPC handlers for start/stop/restart/getStatus added correctly; supervisor events forwarded to renderer via sendSymphonyStatusToRenderer with proper cleanup in return function. |
| apps/desktop/src/main/index.ts | SymphonySupervisor instantiated on app ready and properly stopped on before-quit; clean wiring with registerSessionIpc. |
| apps/desktop/src/preload/index.ts | Symphony API surface (getStatus, start, stop, restart, onStatus) correctly exposed via contextBridge with proper IPC channel mappings. |
| apps/desktop/src/renderer/atoms/symphony.ts | Jotai atoms for symphony status, pending state, and command dispatch; useSymphonyBridge hook fetches initial status and subscribes to live updates correctly. |
| apps/desktop/src/renderer/components/settings/SymphonyRuntimePanel.tsx | Clean UI panel with start/stop/restart controls, phase badge, error display, and config guidance; control-state derivation logic is sound. |
| apps/desktop/src/shared/types.ts | New Symphony types (SymphonyRuntimePhase, SymphonyRuntimeStatus, SymphonyRuntimeError, SymphonyLaunchDescriptor, DesktopApi.symphony) are well-typed and complete. |
| apps/desktop/e2e/tests/symphony-runtime.e2e.ts | E2E tests cover start/stop/restart lifecycle, config error, and readiness failure states via the mock mode fixture; good coverage of the three modes. |
| apps/desktop/src/main/__tests__/symphony-supervisor.test.ts | Comprehensive unit tests for supervisor lifecycle including process starts, diagnostics, spawn errors, stop timeout, and workspace changes. |
| apps/desktop/src/main/__tests__/symphony-config.test.ts | Thorough tests for all resolveSymphonyLaunch paths: preferences, env fallbacks, binary discovery, and edge cases like quoted values and inline comments. |
| apps/desktop/e2e/fixtures/electron.fixture.ts | symphonyMockMode fixture option added with default 'ready'; KATA_DESKTOP_SYMPHONY_MOCK and KATA_SYMPHONY_URL env vars correctly injected into Electron process. |

</details>

</details>

<h3>Sequence Diagram</h3>

```mermaid
sequenceDiagram
    participant R as Renderer
    participant P as Preload (contextBridge)
    participant I as IPC Main
    participant S as SymphonySupervisor
    participant C as Child Process (symphony)

    Note over R,C: Initial Load
    R->>P: window.api.symphony.getStatus()
    P->>I: ipcMain symphony:get-status
    I->>S: getStatus()
    S-->>I: SymphonyRuntimeStatus {phase:'idle'}
    I-->>P: SymphonyRuntimeStatusResponse
    P-->>R: status atom updated

    Note over R,C: User clicks Start
    R->>P: window.api.symphony.start()
    P->>I: ipcMain symphony:start
    I->>S: start()
    S->>S: resolveSymphonyLaunch()
    S->>C: spawn(command, args)
    S-->>I: status push {phase:'starting'}
    I->>P: webContents.send symphony:status
    P->>R: onStatus callback -> atom update
    S->>C: GET /api/v1/state (readiness poll)
    C-->>S: 200 OK
    S-->>I: CommandResult {success:true, phase:'ready'}
    S->>I: emit status {phase:'ready'}
    I->>P: webContents.send symphony:status
    P->>R: atom update -> badge shows Ready
    I-->>P: IPC reply
    P-->>R: runSymphonyCommandAtom resolves

    Note over R,C: User clicks Stop
    R->>P: window.api.symphony.stop()
    P->>I: ipcMain symphony:stop
    I->>S: stop()
    S->>C: kill(SIGTERM)
    C-->>S: exit event
    S-->>I: CommandResult {success:true, phase:'stopped'}
    I-->>P: IPC reply

    Note over R,C: App quit
    P->>I: app before-quit
    I->>S: stop('app_quit')
    S->>C: kill(SIGTERM)
    C-->>S: exit
    I->>I: bridge.shutdown()
    I->>I: app.exit(0)
```

<details><summary>Prompt To Fix All With AI</summary>

`````markdown
This is a comment left during a code review.
Path: apps/desktop/src/main/symphony-supervisor.ts
Line: 298-316

Comment:
**SIGKILL outcome ignored — supervisor always enters `failed` state even after a successful forceful kill**

After SIGKILL is sent, the result of the second `waitForExit(child, 1_000)` is discarded. The function unconditionally returns `{ success: false }` with a `STOP_TIMEOUT` error and sets `phase: 'failed'`, regardless of whether SIGKILL actually terminated the process.

In practice, SIGKILL succeeds in the vast majority of cases. This means:
- The process is dead, the child reference is nulled, `pid` is cleared — the runtime is in a **clean state**.
- But the UI shows `Failed` with a "Timed out" error, and `success: false` is returned to the caller.
- A subsequent `start()` call from a confused user would then try to spawn a fresh process, which is correct — but they first have to dismiss a spurious error.

The fix is to check whether the SIGKILL wait succeeded and, if so, transition to `'stopped'` rather than `'failed'`.

How can I resolve this? If you propose a fix, please make it concise.

---

This is a comment left during a code review.
Path: apps/desktop/src/main/symphony-config.ts
Line: 13-15

Comment:
**Inverted env-var constant naming**

The variable names are backwards relative to their actual roles and priority in the lookup chain:

- `FALLBACK_URL_ENV_KEY` holds `'KATA_SYMPHONY_URL'` — this is checked **first** (the preferred/newer key), but the name says "FALLBACK".
- `DEFAULT_URL_ENV_KEY` holds `'SYMPHONY_URL'` — this is the legacy value tried only when `KATA_SYMPHONY_URL` is absent, but the name says "DEFAULT".

The test at line 498 confirms the intent: _"falls back to SYMPHONY_URL when KATA_SYMPHONY_URL is missing"_ — `SYMPHONY_URL` is the fallback, not `KATA_SYMPHONY_URL`. Consider renaming the constants to `PRIMARY_URL_ENV_KEY` / `LEGACY_URL_ENV_KEY` (or similar) to match the actual lookup order and prevent future misreads.

How can I resolve this? If you propose a fix, please make it concise.
`````

</details>

<sub>Reviews (1): Last reviewed commit: ["test(desktop): KAT-2340 cover symphony c..."](https://github.com/gannonh/kata/commit/c770eede9405ffd3ca94472a897a28eac38d59ae) | [Re-trigger Greptile](https://app.greptile.com/api/retrigger?id=27361561)</sub>

> Greptile also left **2 inline comments** on this PR.

### CODE REVIEW

File: apps/desktop/src/main/symphony-supervisor.ts [hunk 1]
```diff
@@ -0,0 +1,534 @@
...
+        managedProcessRunning: true,
+        pid: child.pid ?? null,
+      })
+
+      const readiness = await this.waitForReadiness(launch.resolvedUrl)
+      if (!readiness.ok) {
+        this.updateStatus({
+          phase: 'failed',
+          managedProcessRunning: Boolean(this.child),
+          lastError: readiness.error,
```
[the remark this answers is not in the document]
Comment (gannonh) [reply apps/desktop/src/main/symphony-supervisor.ts:246]: Fixed in `81d4dc97`.

On the readiness-failure path we now explicitly tear down the managed child process before returning failure:
- added `terminateChildAfterReadinessFailure()` to send `SIGTERM`, escalate to `SIGKILL` on timeout, and clear the child handle.
- call this cleanup when `waitForReadiness()` returns `READINESS_FAILED`.

Also added regression assertions in `symphony-supervisor.test.ts` to verify readiness failure leaves `managedProcessRunning=false` and sends `SIGTERM`.

File: apps/desktop/src/main/symphony-supervisor.ts [hunk 2]
```diff
@@ -0,0 +1,534 @@
...
+    return { success: true, status: this.status }
+  }
+
+  private async waitForReadiness(url: string): Promise<{ ok: true } | { ok: false; error: SymphonyRuntimeError }> {
+    const startedAt = Date.now()
+
+    while (Date.now() - startedAt < this.readinessTimeoutMs) {
+      const endpoint = buildEndpoint(url, '/api/v1/state')
+      this.updateStatus({ lastReadinessCheckAt: new Date().toISOString() })
+
```
[the remark this answers is not in the document]
Comment (gannonh) [reply apps/desktop/src/main/symphony-supervisor.ts:455]: Addressed in `81d4dc97`.

`waitForReadiness()` now short-circuits when the spawned child is gone (`!this.child` or `exitCode !== null`) instead of waiting for the full readiness timeout. In that case it returns the process-exit error context (`status.lastError` when available, otherwise a `PROCESS_EXITED` fallback).

Added a regression test (`returns process exit error when child exits before readiness succeeds`) that simulates early child exit and verifies we return `PROCESS_EXITED` with only a single readiness fetch attempt.

File: apps/desktop/src/main/symphony-supervisor.ts [hunk 3]
```diff
@@ -0,0 +1,534 @@
...
+      }
+
+      this.updateStatus({
+        phase: 'failed',
+        managedProcessRunning: false,
+        pid: null,
+        lastError: error,
+      })
+
+      return { success: false, status: this.status, error }
```
[the remark this answers is not in the document]
Comment (gannonh) [reply apps/desktop/src/main/symphony-supervisor.ts:343]: Good catch — fixed in `2161d3b1`.

`stopInternal()` now checks the post-`SIGKILL` wait result:
- if `SIGKILL` leads to exit within the bounded wait, we transition to `phase: 'stopped'` and return `success: true`.
- only if the process still does not exit after `SIGKILL` do we keep `STOP_TIMEOUT` and return failure.

I also added `treats SIGKILL exit as successful stop after graceful timeout` in `symphony-supervisor.test.ts` to lock this behavior.

File: apps/desktop/src/main/symphony-config.ts [hunk 4]
```diff
@@ -0,0 +1,415 @@
...
+import type {
+  SymphonyConfigSource,
+  SymphonyLaunchDescriptor,
+  SymphonyLaunchSource,
+  SymphonyRuntimeError,
+} from '../shared/types'
+
+const DEFAULT_URL_ENV_KEY = 'SYMPHONY_URL'
+const FALLBACK_URL_ENV_KEY = 'KATA_SYMPHONY_URL'
+const SYMPHONY_BIN_ENV_KEY = 'KATA_SYMPHONY_BIN_PATH'
```
[the remark this answers is not in the document]
Comment (gannonh) [reply apps/desktop/src/main/symphony-config.ts]: Agreed — renamed in `2161d3b1` for clarity.

Updated the env URL constants to reflect actual precedence:
- `PRIMARY_URL_ENV_KEY = 'KATA_SYMPHONY_URL'`
- `LEGACY_URL_ENV_KEY = 'SYMPHONY_URL'`

Also updated the lookup and error message references so naming now matches behavior (`KATA_SYMPHONY_URL` first, legacy fallback second).

File: apps/desktop/src/main/symphony-config.ts [hunk 5]
```diff
@@ -0,0 +1,415 @@
...
+      typeof error === 'object' && error !== null && 'code' in error
+        ? String((error as { code?: unknown }).code)
+        : undefined
+
+    if (code === 'ENOENT') {
+      return null
+    }
+
+    return null
+  }
```
[the remark this answers is not in the document]
Comment (gannonh) [reply apps/desktop/src/main/symphony-config.ts:117]: Addressed in `d2c9b0f2`.

`loadWorkspacePreferences()` now only treats `ENOENT` as "missing". For all other read errors it throws, and `resolveSymphonyLaunch()` converts that into a typed `CONFIG_INVALID` (`details: preferences_read_failed`) so the real failure is surfaced instead of silently falling through.

File: apps/desktop/src/main/symphony-config.ts [hunk 6]
```diff
@@ -0,0 +1,415 @@
...
+      },
+    }
+  }
+
+  if (options.appIsPackaged) {
+    const packagedPath = path.join(options.resourcesPath ?? process.resourcesPath, 'symphony')
+    if (isExecutableFile(packagedPath)) {
+      return { ok: true, command: packagedPath, source: 'bundled' }
+    }
+  }
```
[the remark this answers is not in the document]
Comment (gannonh) [reply apps/desktop/src/main/symphony-config.ts:295]: Fixed in `d2c9b0f2`.

Implemented cross-platform path/binary handling updates:
- packaged lookup now checks `symphony.exe` first on Windows, then `symphony`.
- `toAbsolutePath()` now normalizes backslashes before resolving, and still expands `~`.
- `isExecutableFile()` now probes both `<path>` and `<path>.exe` on Windows.

Also added config tests covering path normalization and preference read error surfacing.

File: apps/desktop/src/main/symphony-supervisor.ts [hunk 7]
```diff
@@ -0,0 +1,534 @@
...
+  }
+
+  public async setWorkspacePath(workspacePath: string): Promise<void> {
+    const resolved = workspacePath.trim()
+    if (!resolved || resolved === this.workspacePath) {
+      return
+    }
+
+    await this.stop('workspace_changed')
+    this.workspacePath = resolved
```
[the remark this answers is not in the document]
Comment (gannonh) [reply apps/desktop/src/main/symphony-supervisor.ts]: Implemented in `d2c9b0f2`.

Added a shared lifecycle gate (`withLifecycleGate`) and routed `start()`, `stop()`, `restart()`, and `setWorkspacePath()` through it so lifecycle operations serialize instead of interleaving across workspaces.

Also updated internal sequencing to avoid re-entrant deadlocks (e.g. restart path now uses `stopInternal('restart')` while already gated).

File: apps/desktop/src/main/symphony-supervisor.ts [hunk 8]
```diff
@@ -0,0 +1,534 @@
...
+          managedProcessRunning: Boolean(this.child),
+          lastError: readiness.error,
+          lastReadinessCheckAt: new Date().toISOString(),
+        })
+
+        return {
+          success: false,
+          status: this.status,
+          error: readiness.error,
+        }
```
[the remark this answers is not in the document]
Comment (gannonh) [reply apps/desktop/src/main/symphony-supervisor.ts:254]: Addressed across `81d4dc97`, `2161d3b1`, and `d2c9b0f2`.

Startup/readiness failure behavior is now unified and no longer leaves orphan processes or masks earlier failures:
- readiness timeout path explicitly tears down the child (`terminateChildAfterReadinessFailure`).
- readiness polling short-circuits when child exits and returns process-exit context instead of waiting full timeout.
- lifecycle operations are serialized via `withLifecycleGate` to prevent cross-operation races during startup/stop/workspace transitions.

Regression coverage was added for readiness failure cleanup and early child-exit behavior in `symphony-supervisor.test.ts`.
